push_test.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	smartling "github.com/Smartling/api-sdk-go"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+type request struct {
+	response string
+	code     int
+}
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func mockHttpClient(fn roundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: fn,
+	}
+}
+
+func TestPushStopUnauthorized(t *testing.T) {
+	args := getArgs("README.md README.md")
+
+	httpClient := getMockHttpClient([]request{{"{}", 401}})
+
+	mockGlobber(args)
+	defer func() {
+		globFilesLocally = globFilesLocallyFunc
+	}()
+
+	client := getClient(httpClient)
+
+	err := doFilesPush(&client, getConfig(), args)
+
+	assert.EqualError(
```

Comment:
Can we make asserts in a more classic style? Like `client.UploadFile()` must be called only once.